assets/js/index.js

```diff
@@ -0,0 +1,146 @@
+//~ Declarations
+
+// declaring each working hour in a variable using an object array
+const workingHours = [
+  {
+    timeLabel: "09:00AM",
+    key: 9,
+  },
+  {
+    timeLabel: "10:00AM",
+    key: 10,
+  },
+  {
+    timeLabel: "11:00AM",
+    key: 11,
+  },
+  {
+    timeLabel: "12:00AM",
+    key: 12,
+  },
+  {
+    timeLabel: "13:00PM",
+    key: 13,
+  },
+  {
+    timeLabel: "14:00PM",
+    key: 14,
+  },
+  {
+    timeLabel: "15:00PM",
+    key: 15,
+  },
+  {
+    timeLabel: "16:00PM",
+    key: 16,
+  },
+  {
+    timeLabel: "17:00PM",
+    key: 17,
+  },
+];
+
+//~ LOCAL STORAGE
+
+const readFromLocalStorage = (key) => {
+  // get from LS using key name
+  const dataFromLS = localStorage.getItem(key);
+
+  // parse data from LS
+  const parsedData = JSON.parse(dataFromLS);
+  if (parsedData) {
+    return parsedData;
+  } else {
+    return "";
+  }
+};
+
+const writeToLocalStorage = (key, value) => {
+  // convert value to string
+  const stringifiedValue = JSON.stringify(value);
+
+  // set stringified value to LS for key name
+  localStorage.setItem(key, stringifiedValue);
+};
+
+//~ FUNCTIONS SECTION
+
+// function to create HTML for each time block, append to page
+const renderTimeBlocks = () => {
+  // target the time block container and create time blocks for each working hour in workingHours array
+  $.each(workingHours, function (index, workingHour) {
+    // get current hour using moment js
+    const currentHour = moment().hour();
+    // function to render the textarea color depending on the conditional statements
+    const renderTextareaColor = () => {
```

Comment:
You could have declared this function outside the scope of the each iterator so that the function is not recreated for every time block. Something to consider in the future